samtranslator/model/lambda_.py

```diff
@@ -1,38 +1,37 @@
 from typing import Any, Dict, List, Optional, Union
 
-from samtranslator.model import PassThroughProperty, PropertyType, Resource
+from samtranslator.model import PassThroughProperty, Resource
 from samtranslator.model.intrinsics import fnGetAtt, ref
-from samtranslator.model.types import IS_DICT, IS_STR, any_type, is_type, list_of, one_of
 from samtranslator.utils.types import Intrinsicable
 
 
 class LambdaFunction(Resource):
     resource_type = "AWS::Lambda::Function"
     property_types = {
-        "Code": PropertyType(True, IS_DICT),
-        "PackageType": PropertyType(False, IS_STR),
-        "DeadLetterConfig": PropertyType(False, IS_DICT),
-        "Description": PropertyType(False, IS_STR),
-        "FunctionName": PropertyType(False, IS_STR),
-        "Handler": PropertyType(False, IS_STR),
-        "MemorySize": PropertyType(False, is_type(int)),
-        "Role": PropertyType(False, IS_STR),
-        "Runtime": PropertyType(False, IS_STR),
-        "Timeout": PropertyType(False, is_type(int)),
-        "VpcConfig": PropertyType(False, IS_DICT),
-        "Environment": PropertyType(False, IS_DICT),
-        "Tags": PropertyType(False, list_of(IS_DICT)),
-        "TracingConfig": PropertyType(False, IS_DICT),
-        "KmsKeyArn": PropertyType(False, one_of(IS_DICT, IS_STR)),
-        "Layers": PropertyType(False, list_of(one_of(IS_STR, IS_DICT))),
-        "ReservedConcurrentExecutions": PropertyType(False, any_type()),
-        "FileSystemConfigs": PropertyType(False, list_of(IS_DICT)),
-        "CodeSigningConfigArn": PropertyType(False, IS_STR),
-        "ImageConfig": PropertyType(False, IS_DICT),
-        "Architectures": PropertyType(False, list_of(one_of(IS_STR, IS_DICT))),
-        "SnapStart": PropertyType(False, IS_DICT),
-        "EphemeralStorage": PropertyType(False, IS_DICT),
-        "RuntimeManagementConfig": PropertyType(False, IS_DICT),
+        "Code": PassThroughProperty(True),
```

Comment:
These can all just be `GeneratedProperty()`. We don't need to do any runtime validation.

Reply:
+1

Reply:
Oh right! Yes `GeneratedProperty()` is better.